### PR TITLE
Life improvement: Use nav.html and load with nav.js

### PR DESCRIPTION
**Pull request by milktea02**
I saw #3 and #4 and figured I'd try find a way to simplify the nav bar generation that might be easy to understand and maintain.

**Proposed change:**
- Store the sidebar items in `nav.html`
- Load `nav.html` into each page using `nav.js` script
- Based on this stackoverflow  solution  using pure html and javascript
  - https://stackoverflow.com/a/68909928/5814222

**Why?**
- Easier to change on `nav.html` document than copy past across all your html files

**Testing:**
- You'll likely need to enable CORS if running on chrome (and possibly other browsers?) or start a simple http server
  - Why? Because when running locally for testing, `nav.html` is coming from your filesystem (eg. `file://localhost/path/to/nav`) which is different from `index.html` which loads on your browser as `localhost/path/to/index.hml` and your browser thinks that's shady
  - You can use chrome extension to run an http server (https://chrome.google.com/webstore/detail/web-server-for-chrome/ofhbbkphhbklhfoeikjpcbhemlocgigb/)
  - or if you're familiar with the terminal and python you can use `python -m http.server`

Tada!
image

### CODE REVIEW

**Comment by 5HT2** [comment]
Do we already use JS? I'd rather not use JS in a static site that doesn't use any. 

Using a shared file isn't a bad idea - I was in the process of setting up templating for i18n and it wouldn't be hard to split components into parts.

**Comment by solonovamax** [comment]
If you wanted to do templating, you could consider something like how I have [my site](https://solonovamax.gay) set up ([source code](https://github.com/solonovamax/solonovamax.gay/))

Basically, I have a couple of thymeleaf templates set up (that inherit from some layout templates), which will inject markdown files into the html. That way, I can write content in markdown, and then it will be injected into the site when requested by a user. (It is cached after the first request, to avoid long load times. Caching can be disabled for development)

My site is written entirely without a single line of js (I hate it with a passion), so you don't have to worry about that. It's 100% server side.

Also, I have a few other fancy things, like emojis in markdown (using custom flexmark extension to allow for arbitrary new emojis not in the original spec), a config file for loading contact links, and split frontend/backend so that they can be updated independently

**Comment by 5HT2** [comment]
> My site is written entirely without a single line of js (I hate it with a passion), so you don't have to worry about that. It's 100% server side.

I feel similarly, though I would've setup templating by now if this wasn't entirely static. I would prefer QTC templates and have a github action to do the static generation after a push? Honestly a pain in the ass to setup regardless, but eventually more ideal than what we have going on or a non-serverless solution.

**Comment by 5HT2** [comment]
@milktea02

Sorry for the incredibly late reply - I've been on vacation for a few weeks and was just able to get back to this.

Would you mind rebasing to the main branch and fixing the conflicts? I'll be free to look over it whenever (any hours outside of 09:00 to 15:00 PDT).

**Comment by milktea02** [comment]
@5HT2 

ah sorry, I've also been MIA - I will try get the rebase done this week


As for earlier questions/comments: 

I agree, the website wouldn't be fully static with the addition of JS. I also prefer templated solutions but as mentioned already, the site is already fully static so this seemed to be a less invasive solution.

Feel free to disregard this PR if it's not in the direction the maintainers prefer.

**Comment by 5HT2** [comment]
I don't mind JS being used for this, please feel free to rebase onto the main branch and fix the PR to be mergeable, it would be greatly appreciated :)

File: nav.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+fetch('nav.html')
+.then(res => res.text())
+.then(text => {
+    let oldelem = document.querySelector("script#replace_with_navbar");
+    let newelem = document.createElement("div");
+    newelem.innerHTML = text;
+    oldelem.parentNode.replaceChild(newelem,oldelem);
```
Comment (5HT2) [inline nav.js:7]: snakeCase for these please, I'll fix it myself

**Review by 5HT2 — CHANGES_REQUESTED** [review]

**Review by 5HT2 — APPROVED** [review]
Thanks, looks good to me!

**Pull request merged**